commons-hpcc/src/main/java/org/hpccsystems/commons/errors/HpccErrorBlock.java

```diff
@@ -0,0 +1,154 @@
+package org.hpccsystems.commons.errors;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+/**
+ * Store a set of HpccErrorr.
+ *
+ */
+
+public class HpccErrorBlock
+{
+
+    private final CopyOnWriteArraySet<HpccError> he = new CopyOnWriteArraySet<>();
+
+    public HpccErrorBlock()
+    {
+
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see java.lang.Object#toString()
+     */
+    @Override
+    public String toString()
+    {
+        StringBuilder s = new StringBuilder();

```

Comment:
consider synchronizing

Reply:
Not necessary as the internal array  is a copyOnWriteArraySet (see https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CopyOnWriteArraySet.html for details)